main.go

```diff
@@ -122,7 +123,17 @@ func main() {
 		secretsDir:         *secretsDir,
 		metadataLabels:     nodeMetadata,
 		deploymentInfo:     deploymentInfo,
-	})
+		configScope:        *configScope,
+	}
+
+	validationErr := validate(input)
```

Comment:
Combine this assignment with the if, so that the variable goes out of scope? Err... actually, this is just a normal error, so leave it as-is but rename the variable to `err`.